Foundation/URLSession/URLSessionTask.swift

```diff
@@ -428,7 +428,21 @@ open class URLSessionDownloadTask : URLSessionTask {
      * If resume data cannot be created, the completion handler will be
      * called with nil resumeData.
      */
-    open func cancel(byProducingResumeData completionHandler: @escaping (Data?) -> Void) { NSUnimplemented() }
+    open func cancel(byProducingResumeData completionHandler: @escaping (Data?) -> Void) {
+        super.cancel()
+        
+        /*
+         * In Objective-C, this method relies on an Apple-maintained XPC process
+         * to manage the bookmarking of partially downloaded data. Therefore, the
+         * original behavior cannot be directly ported, here.
+         *
+         * Instead, we just call the completionHandler directly.
+         *
+         * A token Data value is passed to the client to prevent unexpected failures
```

Comment:
This would be prudent if data was _usually_ expected, but [the contract of this method goes](https://developer.apple.com/documentation/foundation/urlsessiondownloadtask/1411634-cancel):

> If the download is resumable, the completion handler is provided with a resumeData object. Your app can later pass this object to a session’s downloadTask(withResumeData:) or downloadTask(withResumeData:completionHandler:) method to create a new task that resumes the download where it left off.

Since we do not yet support resuming on Linux, no downloads are resumable, and so we can safely pass `nil` here. Let's not confuse downstream clients into thinking the download is resumable when we can't resume it.

Reply:
Great point. PR updated, accordingly.